File: david/xgboost/get_sepsis_score.R
#!/usr/bin/Rscript

get_sepsis_score = function(data, model){
    require(magrittr)
    # data <- current %>% 
    #     as.data.frame() %>% 
    #     select(-one_of("subject", "SepsisLabel", "EtCO2", "Bilirubin_direct"))
    # xgb.mat <- xgb.DMatrix(data = data.matrix(current_data))
    
    # data_last_row <- data[nrow(data),]
    
    score <- predict(model, data)[nrow(data)]
    
    # score = 1 - exp(-l_exp_bx)
    label = score > 0.1
    predictions = c(score, label)
    return(predictions)
}

load_sepsis_model = function(){
    model <- xgboost::xgb.load(here::here("david", "xgboost", "xgboost.model"))
    return(model)
}

File: david/get_sepsis_score.py
#!/usr/bin/env python

import xgboost as xgb

def get_sepsis_score(data, model):

    # Get number of rows in data to know which prediction to save
    row_total = data.shape[0]

    dtest = xgb.DMatrix(data)

    # score = 1 - np.exp(-l_exp_bx)
    score = model.predict(dtest)[row_total-1]
    label = score > 0.02

    return score, label

def load_sepsis_model():

    model = xgb.Booster(model_file="xgboost/xgb.model")

    return model
